collections/_posts/2022-10-24-HierarchicalProbabilisticUNet.md

```diff
@@ -0,0 +1,198 @@
+
+
+# Method
+
+## Appetizer
+
+The following animation illustrates the capacity of the method in modelling complex structured distributions across scales.
+
+![](/collections/images/hierarchical_probabilistic_unet/animation.gif)
+
+&nbsp;
+
+## Architecture
+
+* The architecture is based on the ***conditional VAE*** whose details are provided in the following [tutorial](https://creatis-myriad.github.io/tutorials/2022-09-12-tutorial-cvae.html).
+* The main innovation comes from the following hierarchical modelling of the latent space:
+
+$$p\left(\boldsymbol{z} \vert x\right) = p\left(z_0,\ldots,z_L \vert x\right) = p\left(z_L \vert z_{<L},x\right) \cdot \, \ldots \, \cdot p\left(z_0 \vert x\right)$$
+
+$$q\left(\boldsymbol{z} \vert x,y\right) = q\left(z_0,\ldots,z_L \vert x,y\right) = q\left(z_L \vert z_{<L},x,y\right) \cdot \, \ldots \, \cdot q\left(z_0 \vert x,y\right)$$
+
+In the particular case where $$L=2$$, the above equation can be put in the following form:
+
+$$p\left(\boldsymbol{z} \vert x\right) = p\left(z_2 \vert z_1,z_0,x\right) \cdot p\left(z_1 \vert z_0,x\right) \cdot p\left(z_0 \vert x\right)$$
+
+$$q\left(\boldsymbol{z} \vert x,y\right) = q\left(z_2 \vert z_1,z_0,x,y\right) \cdot q\left(z_1 \vert z_0,x,y\right) \cdot q\left(z_0 \vert x,y\right)$$
+
+* Taking into account the hierarchical modelling, a new ELBO objective with a relative weighting factor $$\beta$$ was formulated as follows (the corresponding demonstration is given at the end of this post):
+
+$$\mathcal{L}_{ELBO} = \mathbb{E}_{\boldsymbol{z}\sim q(\boldsymbol{z} \vert x,y)} [CE\left( y,\hat{y}\right)] + \beta \cdot \sum_{i=0}^{L} \mathbb{E}_{z_{i-1}\sim \prod_{j=0}^{i-1} q(z_j \vert z_{<j},x,y)} [D_{KL}(q(z_i \vert z_{<i},x,y) \parallel p(z_i \vert z_{<i},x))]$$
+
+* The authors observed that the minimization of $$\mathcal{L}_{ELBO}$$ leads to sub-optimally results. For this reason, they used the recently proposed $$GECO$$ loss:
+
+$$\mathcal{L}_{GECO} = \lambda \cdot \left( \mathbb{E}_{\boldsymbol{z}\sim q(\boldsymbol{z} \vert x,y)} [CE\left( y,\hat{y}\right)] - \kappa \right) + \sum_{i=0}^{L} \mathbb{E}_{z_{i-1}\sim \prod_{j=0}^{i-1} q(z_j \vert z_{<j},x,y)} [D_{KL}(q(z_i \vert z_{<i},x,y) \parallel p(z_i \vert z_{<i},x))]$$
+
+where $$\kappa$$ is chosen as the desired reconstruction error and $$\lambda$$ is a Lagrange multiplier that is updated as a function of the ***exponential moving average*** of the reconstruction contraint.
+
+> This formulation initially puts high pressure on the reconstruction and once the desired $$\kappa$$ is reached it increasingly moves the pressure over on the KL-terms.
+
+* Finally, the prior and the generator networks are based on the same U-Net architecture, which results in parameter and run-time savings.
+
+The overall architecture is given below:
+
+![](/collections/images/hierarchical_probabilistic_unet/overall_architecture.jpg)
```

Comment:
If we want to be really nitpicky, I don't think the definition of _y_ is accurate, since _y_ is more constrained than a real.

So, instead of _y \in R^{N \times N}_, I would rather define something like this: _y \in [0, C]^{N \times N}_.

P.S. This comment also applies to definitions of the ground truth in later figures.

Reply:
I would have made the modification for an article, here I will leave like that to avoid me to remake the figures ;) We can mention it during the oral presentation

Reply:
Works for me :wink: